### PR TITLE
[bug] live reload fixed

### PR DESCRIPTION
**Pull request by jexroid**
By submitting this pull request, I confirm that my contribution is made under the terms of the MIT license.

## Problem/Feature

air live reload will be working on both Unixbased and Windows machines

## Description of Changes: 

- Related to issue ticket #282 

1. `UnixBased` with bool type is added to `Project` type
2. CheckOs() function is added for checking the UNIX-based operating system if it's not windows leaves the `UnixBased` to false by default
```go
func (p *Project) CheckOs() {
	if runtime.GOOS != "windows" {
		p.UnixBased = true
	}
}
```
3. CheckOs() is executing in program.go:260
```go
// Define Operating system
p.CheckOs()
```
4. makefile.tmpl and air.toml.tmpl are modified

## Checklist

- [x] I have self-reviewed the changes being requested
- [ ] I have updated the documentation (check issue ticket #218)

### CODE REVIEW

**Comment by Ujstor** [comment]
@jexroid 
Nice PR! This solves the issue of manually configuring Air to get it working in Windows. Is there a way we can get the make watch command to work in Win? Check if Air is installed and offer an installation option. Is there a PowerShell syntax that will do the same as in Unix-based systems? 

If not, we can use templating and just map make watch to air with a comment that Air can be installed with go install github.com/air-verse/air@latest

**Comment by jexroid** [comment]
thanks @Ujstor I think im able to write a PowerShell for that which checks if air command return system error or not found. and return error statement that tell the user to install it your self or do the instructions for installing it using `GO get`

**Comment by jexroid** [comment]
these failing tests are because of this:
```
{{if and (ne .DBDriver "none") (ne .DBDriver "sqlite")}}
# Integrations Tests for the application
itest:
	@echo "Running integration tests..."
	@go test ./internal/database -v
{{end}}
```

and the CI is testing for itest which will never exist in a project with no database or with sqlite
is there anything else needed @Ujstor

**Comment by Ujstor** [comment]
I am a little bit busy...I will try to check the PR this weekend

**Comment by jexroid** [comment]
alright, as we discussed the purpose of this PR is to generate a binary version of the go project based on the user OS which is configured in Makefile and run the file in .air.toml which solve the hot reloading in windows machines but there will be no Air auto installation on windows but its behave like before for Unix base machines

File: cmd/template/framework/files/makefile.tmpl [hunk 1]
```diff
@@ -51,20 +51,25 @@ clean:
...
+                echo "Watching...";\
+            else \
+                echo "You chose not to install air. Exiting..."; \
+                exit 1; \
+            fi; \
+        fi
+{{else}}
+watch:
+	@air
+{{end}}
```
Comment (Ujstor) [inline cmd/template/framework/files/makefile.tmpl:73]: Indentation is wrong; this is why the tests are failing.

For local tests, run:

go build
go install

Then create the project. Test the features that you are implementing.

File: cmd/template/framework/files/makefile.tmpl [hunk 2]
```diff
@@ -51,20 +51,25 @@ clean:
...
 
 .PHONY: all build run test clean
```
Comment (Ujstor) [inline cmd/template/framework/files/makefile.tmpl:75]: Nit: Update .PHONY with commands that are missing.

**Review by Ujstor — CHANGES_REQUESTED** [review]
Check comments.

**Comment by jexroid** [comment]
seems like the `Makefile:56: *** missing separator.  Stop.` occurs when you use spaces instead of tabs. I realized and fixed that while using Windows but couldn't test it on UNIX machines. the problems have been addressed and fixed. 
- **watch** command has been added to **.PHONY**
- Indentation of MakeFile is fixed

**Review by Ujstor — COMMENTED** [review]
LGTM

**Pull request merged**